### PR TITLE
Issue#27 pycsw single microservice

### LINKED ISSUE #27: pyCSW single microservice
pycsw is currently used as a library inside of GeoNode. To reduce the complexity of the code and improve microservice design pattern, to improve scalability and resource management pycsw setup as a single deployment running pycsw. GeoNode already supports external pycsw service by settings.py parameters

### PR DESCRIPTION
**Pull request by mwallschlaeger**
Closes #27 

## 📑 Description
ref #27

added pycsw microservice inspired by pycsw helm chart (https://github.com/geopython/pycsw/tree/master/docker/helm) with adoption to the style of this helm chart.

Implementation:
- nginx container forwarding traffic to pycsw via  `.Values.pycsw.endpoint` url
- mappings.py taken from GeoNode codbase (https://github.com/GeoNode/geonode/blob/master/geonode/catalogue/backends/pycsw_local_mappings.py)
- pycsw.cfg taken from pycsw helm chart

### CODE REVIEW

**Comment by mwallschlaeger** [comment]
waiting for https://github.com/GeoNode/geonode/pull/11319 . meanwhile you can use the following geonode image including the bugfix promoted inside the PR.
```
  image:
    # -- used geonode image
    name: mwall2bitflow/geonode
    # -- tag of used geonode image
    tag: '4.1.x-pycsw-hotfix'
```

**Comment by mwallschlaeger** [comment]
https://github.com/GeoNode/geonode/pull/11319 got merged, so no reason to hold this back anymore

File: deployment/geonode/templates/pycsw/pycsw-cfg.yaml
Comment (ridoo) [inline deployment/geonode/templates/pycsw/pycsw-cfg.yaml:8]: Should we better include that configuration as is in the values.yml and embed the config via `{{- include "pycsw.config" . | nindent 4 }}`?

File: deployment/geonode/templates/pycsw/pycsw-deploy.yaml
Comment (ridoo) [inline deployment/geonode/templates/pycsw/pycsw-deploy.yaml:5]: #64

File: deployment/geonode/templates/pycsw/pycsw-env.yaml
Comment (ridoo) [inline deployment/geonode/templates/pycsw/pycsw-env.yaml:5]: #64

File: deployment/geonode/templates/pycsw/pycsw-svc.yaml
Comment (ridoo) [inline deployment/geonode/templates/pycsw/pycsw-svc.yaml:5]: #64

File: docs/pycsw.md
Comment (ridoo) [inline docs/pycsw.md:4]: _without_ having to change GeoNode

File: docs/pycsw.md
Comment (ridoo) [inline docs/pycsw.md:8]: Would be nice to have a link reference here, e.g. https://docs.pycsw.org/en/2.6.1/configuration.html

File: minikube-values.yaml
Comment (ridoo) [inline minikube-values.yaml:34]: Really the default?

**Review by ridoo — CHANGES_REQUESTED** [review]

**Review by ridoo — CHANGES_REQUESTED** [review]
We do not have to resolve #64 first, but at least should have a look at the way how the config is embedded in the configMap

File: deployment/geonode/templates/pycsw/pycsw-deploy.yaml
Comment (mwallschlaeger) [reply deployment/geonode/templates/pycsw/pycsw-deploy.yaml:5]: lets work on #64 in another PR

File: deployment/geonode/templates/nginx/nginx-conf.yaml
Comment (mwallschlaeger) [reply deployment/geonode/templates/nginx/nginx-conf.yaml:133]: lets work on #64 in another PR

File: deployment/geonode/templates/geonode/geonode-env.yaml
Comment (mwallschlaeger) [reply deployment/geonode/templates/geonode/geonode-env.yaml:133]: lets work on #64 in another PR

File: deployment/geonode/templates/_helpers.tpl
Comment (mwallschlaeger) [reply deployment/geonode/templates/_helpers.tpl:21]: lets work on #64 in another PR

File: deployment/geonode/templates/pycsw/pycsw-env.yaml
Comment (mwallschlaeger) [reply deployment/geonode/templates/pycsw/pycsw-env.yaml:5]: lets work on #64 in another PR

**Pull request closed without merging**
